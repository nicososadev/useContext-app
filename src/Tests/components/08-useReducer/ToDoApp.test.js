import React from 'react'
import '@testing-library/jest-dom'
import { mount, shallow } from 'enzyme'
import { ToDoApp } from '../../../components/08-useReducer/ToDoApp'
import { demoToDos } from '../../fixtures/demoToDos'
import { act } from '@testing-library/react'

describe('Pruebas en el componente ToDoApp', () => {
    
    const wrapper = shallow( <ToDoApp />)

    Storage.prototype.setItem = jest.fn()

    test('Debe mostrarse correctamente', () => {
    
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe agregar un toDo y guardarlo en localStorage', () => {
        
        const wrapper = mount( <ToDoApp />)

        act( () => {
            
            const handleAddFunction = wrapper.find('ToDoAdd').prop('handleAdd')

            handleAddFunction( demoToDos[0] )
            handleAddFunction( demoToDos[1] )
        })

        const titleCounter = wrapper.find('h2').text().trim()

        expect( titleCounter ).toBe( 'ToDo List - 2' )

        expect( localStorage.setItem ).toHaveBeenCalledTimes(2)
    })

    test('Debe borrar un toDo', () => {

        act( () => {
            
            const handleAddFunction = wrapper.find('ToDoAdd').prop('handleAdd')
            const handleAddDelete = wrapper.find('ToDoList').prop('handleDelete')

            handleAddFunction( demoToDos[0] )
            handleAddDelete( demoToDos[0].id )
        })

        const titleCounter = wrapper.find('h2').text().trim()

        expect( titleCounter ).toBe( 'ToDo List - 0' )
    })
    
})
