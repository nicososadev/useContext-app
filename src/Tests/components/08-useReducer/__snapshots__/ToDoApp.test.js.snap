// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ToDoApp Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    ToDo App
  </h1>
  <hr />
  <h2>
    ToDo List - 
    0
  </h2>
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <ToDoList
        handleComplete={[Function]}
        handleDelete={[Function]}
        toDos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <ToDoAdd
        handleAdd={[Function]}
      />
    </div>
  </div>
</div>
`;
